Src/DependencyCollector/Shared/Implementation/ProfilerHttpProcessing.cs

```diff
@@ -387,25 +385,33 @@ private void OnEnd(object exception, object thisObj, object returnValue)
                         {
                             responseObj = webException.Response as HttpWebResponse;
                         }
-#if !NET40
-                        if (responseObj == null)
-                        {
-                            var httpException = exception as HttpException;
+                    }
 
-                            if (httpException != null)
-                            {
-                                statusCode = httpException.GetHttpCode();
-                            }
+                    if (responseObj != null)
+                    {
+                        try
+                        {
+                            var statusCode = (int)responseObj.StatusCode;
+                            telemetry.ResultCode = statusCode > 0 ? statusCode.ToString(CultureInfo.InvariantCulture) : string.Empty;
+                            telemetry.Success = (statusCode > 0) && (statusCode < 400);
+                        }
+                        catch (ObjectDisposedException)
+                        {
+                            // ObjectDisposedException is expected here in the following sequence: httpWebRequest.GetResponse().Dispose() -> httpWebRequest.GetResponse()
+                            // on the second call to GetResponse() we cannot determine the statusCode.
                         }
-#endif
+                    }
+                    else if (exception != null)
+                    {
+                        var webException = exception as WebException;
+                        telemetry.ResultCode = webException.Status.ToString();
```

Comment:
To be safe I think you need to check that webException is not null.
e.g. webException?.Status.ToString();